inference-engine/src/transformations/src/transformations/common_optimizations/conv_mul_fusion.cpp

```diff
@@ -85,23 +85,23 @@ NGRAPH_RTTI_DEFINITION(ngraph::pass::GroupConvolutionMultiplyFusion, "GroupConvo
 ngraph::pass::GroupConvolutionMultiplyFusion::GroupConvolutionMultiplyFusion() {
     MATCHER_SCOPE(GroupConvolutionMultiplyFusion);
     auto input = pattern::any_input();
-    auto weights = ngraph::pattern::any_input();//pattern::has_static_dims({0, 1}) /* has GOIYX layout */);
+    auto weights = ngraph::pattern::any_input(pattern::has_static_dims({0, 1}) /* has GOIYX layout */);
     auto conv = ngraph::pattern::wrap_type<opset4::GroupConvolution>({input, weights}, pattern::consumers_count(1));
     auto mul_const = ngraph::pattern::wrap_type<opset4::Constant>();//pattern::has_static_shape());
     auto mul = ngraph::pattern::wrap_type<opset4::Multiply>({conv, mul_const});
 
     matcher_pass_callback callback = [conv, input, weights, mul, mul_const](pattern::Matcher & m) -> bool {
         const auto & pattern_to_output = m.get_pattern_value_map();
 
-        const auto & m_weights = pattern_to_output.at(weights);
+        auto m_weights = pattern_to_output.at(weights);
         const auto & m_const = pattern_to_output.at(mul_const);
-        const auto & m_input = pattern_to_output.at(input);
         const auto & m_conv = pattern_to_output.at(conv).get_node_shared_ptr();
         const auto & m_mul = pattern_to_output.at(mul).get_node_shared_ptr();
 
-        const auto & G = m_weights.get_partial_shape()[0].get_length();
-        const auto & O = m_weights.get_partial_shape()[1].get_length();
-        const auto & weights_rank = m_weights.get_partial_shape().rank().get_length();
+        const auto & weights_shape = m_weights.get_partial_shape();
+        const auto & G = weights_shape[0].get_length();
```

Comment:
Should we check if group count (`G`) equals to channels count and return `false` if not?